src/Auth0.OidcClient.Core/Auth0ClientBase.cs

```diff
@@ -46,29 +53,13 @@ public Task<LoginResult> LoginAsync(object extraParameters = null)
         }
 
         /// <inheritdoc/>
-        public Task<BrowserResultType> LogoutAsync()
+        public async Task<BrowserResultType> LogoutAsync(bool federated = false, object extraParameters = null)
         {
             Debug.WriteLine($"Using Callback URL ${_options.PostLogoutRedirectUri}. Ensure this is an Allowed Logout URL for application/client ID ${_options.ClientId}.");
 
-            return LogoutAsync(false);
-        }
-
-        private IdentityModel.OidcClient.OidcClient OidcClient
-        {
-            get
-            {
-                return _oidcClient ?? (_oidcClient = new IdentityModel.OidcClient.OidcClient(CreateOidcClientOptions(_options)));
-            }
-        }
-
-        /// <inheritdoc/>
-        public async Task<BrowserResultType> LogoutAsync(bool federated)
-        {
-            var logoutParameters = new Dictionary<string, string>
-            {
-                { "client_id", OidcClient.Options.ClientId },
-                { "returnTo", OidcClient.Options.PostLogoutRedirectUri }
-            };
+            var logoutParameters = AppendTelemetry(extraParameters);
+            logoutParameters["client_id"] = OidcClient.Options.ClientId;
```

Comment:
Probably not something we need to address here but adding the Client ID to the logout URL should be optional. Without it, you use the tenant-level logout URLs.

Reply:
Thanks for spotting that! Will create a PR for it as part of this release.